### DISCUSSION TITLE
(npm)Appium server installation gets timeout due to bad connectivity on MacBook Air M2

### DISCUSSION THREAD

**Discussion by Priyagajula** [npm/feedback#838] [General]
Hardware :
MacBook Air (M2, 2022) 

below Prerequisites installed :
Java JDK 19.0.1 and JRE8
Xcode | 14.2
Xcode Command Line Tools | installed
Homebrew | installed
Carthage | installed
asdf | installed

Action : Tried to install Appium server using below command
npm install -g appium

Actual result : 
0 verbose cli /usr/local/bin/node /usr/local/bin/npm
1 info using npm@8.19.2
2 info using node@v18.12.1
3 timing npm:load:whichnode Completed in 0ms
4 timing config:load:defaults Completed in 1ms
5 timing config:load:file:/usr/local/lib/node_modules/npm/npmrc Completed in 0ms
6 timing config:load:builtin Completed in 0ms
7 timing config:load:cli Completed in 1ms
8 timing config:load:env Completed in 0ms
9 timing config:load:project Completed in 1ms
10 timing config:load:file:/Users/dipshi/.npmrc Completed in 0ms
11 timing config:load:user Completed in 0ms
12 timing config:load:file:/usr/local/etc/npmrc Completed in 0ms
13 timing config:load:global Completed in 0ms
14 timing config:load:validate Completed in 0ms
15 timing config:load:credentials Completed in 0ms
16 timing config:load:setEnvs Completed in 1ms
17 timing config:load Completed in 4ms
18 timing npm:load:configload Completed in 4ms
19 timing npm:load:mkdirpcache Completed in 1ms
20 timing npm:load:mkdirplogs Completed in 1ms
21 verbose title npm install appium
22 verbose argv "install" "--global" "appium"
23 timing npm:load:setTitle Completed in 4ms
24 timing config:load:flatten Completed in 0ms
25 timing npm:load:display Completed in 1ms
26 verbose logfile logs-max:10 dir:/Users/dipshi/.npm/_logs
27 verbose logfile /Users/dipshi/.npm/_logs/2022-12-29T22_32_46_160Z-debug-0.log
28 timing npm:load:logFile Completed in 2ms
29 timing npm:load:timers Completed in 0ms
30 timing npm:load:configScope Completed in 0ms
31 timing npm:load Completed in 14ms
32 timing arborist:ctor Completed in 0ms
33 silly logfile done cleaning log files
34 timing idealTree:init Completed in 2ms
35 timing idealTree:userRequests Completed in 1ms
36 silly idealTree buildDeps
37 silly fetch manifest appium@*
38 silly placeDep ROOT appium@ OK for:  want: *
39 timing idealTree:#root Completed in 295047ms
40 timing idealTree:node_modules/appium Completed in 0ms
41 timing idealTree:buildDeps Completed in 295049ms
42 timing idealTree:fixDepFlags Completed in 1ms
43 timing idealTree Completed in 295054ms
44 timing command:install Completed in 295056ms
45 verbose type system
46 verbose stack FetchError: request to https://registry.npmjs.org/appium failed, reason: connect ETIMEDOUT 2606:4700::6810:1a23:443
46 verbose stack     at ClientRequest.<anonymous> (/usr/local/lib/node_modules/npm/node_modules/minipass-fetch/lib/index.js:130:14)
46 verbose stack     at ClientRequest.emit (node:events:513:28)
46 verbose stack     at TLSSocket.socketErrorListener (node:_http_client:494:9)
46 verbose stack     at TLSSocket.emit (node:events:525:35)
46 verbose stack     at emitErrorNT (node:internal/streams/destroy:151:8)
46 verbose stack     at emitErrorCloseNT (node:internal/streams/destroy:116:3)
46 verbose stack     at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
47 verbose cwd /Users/dipshi
48 verbose Darwin 21.6.0
49 verbose node v18.12.1
50 verbose npm  v8.19.2
51 error code ETIMEDOUT
52 error syscall connect
53 error errno ETIMEDOUT
54 error network request to https://registry.npmjs.org/appium failed, reason: connect ETIMEDOUT 2606:4700::6810:1a23:443
55 error network This is a problem related to network connectivity.
55 error network In most cases you are behind a proxy or have bad network settings.
55 error network
55 error network If you are behind a proxy, please make sure that the
55 error network 'proxy' config is set properly.  See: 'npm help config'
56 verbose exit 1
57 timing npm Completed in 295103ms
58 verbose unfinished npm timer reify 1672353166198
59 verbose unfinished npm timer reify:loadTrees 1672353166199
60 verbose code 1
61 error A complete log of this run can be found in:
61 error     /Users/dipshi/.npm/_logs/2022-12-29T22_32_46_160Z-debug-0.log

Note: I verified no proxy enabled on my machine and also network connection is quite strong and I never had any network issues on my old laptop or any hardware

Please help me to solve this issue as I need to setup Appium automation on my new Mac
Thanks in advance

**Comment by MylesBorins** [comment]
This feedback discussion is for sharing suggestions on improvements to npm, not general support. Please try https://github.community or stack overflow for general programming support
